### PR TITLE
Migrate to @fastify/busboy

### LINKED ISSUE #297: BusBoy replacement?
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

We should consider replacing busboy. It is actually quite buggy and has security issues. @mscdex does not really merge PRs, which would fix various bugs.

e.g.
https://github.com/mscdex/dicer/pull/25
https://github.com/mscdex/dicer/pull/22

Maybe we should write it completely new from scratch. The interfaces used by busboy seem to be a success, so we should keep the interfaces but replace the internals. Maybe with actual Streams instead of EventEmitter as baseclass.

Any thoughts?

### PR DESCRIPTION
**Pull request by kibertoad**
This replaces unsupported original version of `busboy` with our own fork, sporting the holy trifecta of security fixes, performance improvements, and a richer feature-set.
It is a semver minor change with no known breaking changes.

fixes #297

#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by Uzlopak** [comment]
I love it. Literally a drop-in-replacement :)

File: package.json [hunk 1]
```diff
@@ -26,7 +25,7 @@
     "eslint-plugin-node": "^11.1.0",
     "eslint-plugin-promise": "^5.1.0",
     "eslint-plugin-typescript": "^0.14.0",
-    "fastify": "^3.2.1",
+    "fastify": "^3.24.1",
```
Comment (zekth) [inline package.json:28]: Is there a specific reason for this version bump?

Comment (kibertoad) [reply package.json:28]: considering the lack of package-lock, it's a noop anyway, just makes it easier to run ncu.

File: test/index.test-d.ts [hunk 2]
```diff
@@ -45,6 +46,7 @@ const runServer = async () => {
     }, (err) => {
       throw err
     }, {
+      headers: { 'content-type': 'multipart/form-data' },
```
Comment (climba03003) [inline test/index.test-d.ts:49]: Just curious why it is added?

Comment (kibertoad) [reply test/index.test-d.ts:49]: it was always mandatory, updated types reflect that

Comment (Uzlopak) [reply test/index.test-d.ts:49]: the content-type-header is mandatory in the typings

**Review by climba03003 — APPROVED** [review]
LGTM.

**Pull request merged**